Purchases/Public/Purchases.swift

```diff
@@ -1242,27 +1244,92 @@ public extension Purchases {
     @available(macOS, unavailable)
     @available(watchOS, unavailable)
     @available(tvOS, unavailable)
-    @objc func beginRefundRequest(for productID: String,
+    @objc func beginRefundRequest(forProduct productID: String,
                                   completion: @escaping (RefundRequestStatus, Error?) -> Void) {
-        purchasesOrchestrator.beginRefundRequest(for: productID, completion: completion)
+        purchasesOrchestrator.beginRefundRequest(forProduct: productID, completion: completion)
     }
 
     /**
      * Presents a refund request sheet in the current window scene for
-     * the latest transaction associated with the productID
+     * the latest transaction associated with the entitlement ID
      *
-     * - Parameter productID: The productID to begin a refund request for.
+     * - Parameter entitlementID: The entitlementID to begin a refund request for.
      * - Parameter completion: A completion block that is called when the modal is closed.
      * If the request was successful, there will be a `RefundRequestStatus`.
      * Keep in mind the status could be `userCancelled`
-     * If the request was unsuccessful, there will be an `Error`.
+     * If the request was unsuccessful or entitlement could not be found, there will be an `Error`.
+     */
+    @available(iOS 15.0, *)
+    @available(macOS, unavailable)
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    @objc func beginRefundRequest(forEntitlement entitlementID: String,
+                                  completion: @escaping (RefundRequestStatus, Error?) -> Void) {
+        purchasesOrchestrator.beginRefundRequest(forEntitlement: entitlementID, completion: completion)
```

Comment:
well right now, the async version just points back to this one, so then we'd have a loop? i could switch up which function calls into the purchasesOrchestrator, i suppose... do we gain anything from that though?

Reply:
well, you'd be able to remove all of the changes in the `Purchases+async` file, since they're only meant for converting methods with completion blocks into async. 
You'd also make `PurchasesOrchestrator.beginRefundRequest` async, which gets you some niceties: 
- guaranteed calls to completion on all paths
- cleaner types: it's _possible_ to call `completion(nil, nil)`, but with an `async` method it'd be impossible to do that

Reply:
in that case, do i need the completion block version of this function at all?

Reply:
oh, i think i remember you saying they are only auto-generated for objc, right?

Reply:
I _think_ you can make the method `@objc` _and_ `async` at the same time, and that'll transform it into completion blocks for objc

Reply:
you can use the API testers to verify that